File: leetcode/java/LEET108.java
public class LEET108 {
    public static class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode() {}
        TreeNode(int val) { this.val = val; }
        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }
    public static void main(String[] args) {
        sortedArrayToBST(new int[] {-10,-3,0,5,9});
    }
    public static TreeNode sortedArrayToBST(int[] nums) {
        return dfs(nums, 0, nums.length-1);
    }
    public static TreeNode dfs(int[] nums, int start, int end){
        if(start > end){
            return null;
        }

        int mid = (start + end) / 2;
        TreeNode node = new TreeNode(nums[mid]);

        node.left = dfs(nums, start, mid-1);
        node.right = dfs(nums, mid+1, end);

        return node;
    }
}

File: 0_3.leetcode/python/LEET108.py
from typing import *
import collections

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def sortedArrayToBST(nums: List[int]) -> TreeNode:
    if not nums:
        return None
    
    mid = len(nums) // 2

    node = TreeNode(nums[mid])
    node.left = sortedArrayToBST(nums[:mid])
    node.right = sortedArrayToBST(nums[mid+1:])

    return node


def solve():
   sortedArrayToBST(nums = [-10,-3,0,5,9]) 
solve()